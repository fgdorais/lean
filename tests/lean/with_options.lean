example {A : Type} (a b c : A) : a = b → b = c → a = c :=
begin
  intro h₁ h₂,
  with_options [pp.implicit true, pp.notation false] state; state,
  with_options [pp.all true, pp.max_depth 1] state,
  with_options [pp.notation false] state,
  with_options [pp.notation false] (state; state),
  substvars
end

example {A : Type} (a b c : A) : a = b → b = c → a = c :=
begin
  intros,
  with_options [] id, -- error
end
