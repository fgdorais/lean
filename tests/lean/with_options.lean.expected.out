with_options.lean:4:53: proof state
A : Type,
a b c : A,
h₁ : @eq A a b,
h₂ : @eq A b c
⊢ @eq A a c
with_options.lean:4:60: proof state
A : Type,
a b c : A,
h₁ : a = b,
h₂ : b = c
⊢ a = c
with_options.lean:5:45: proof state
A : Type.{l_1},
a b c : A,
h₁ : … b,
h₂ : … c
⊢ … c
with_options.lean:6:35: proof state
A : Type,
a b c : A,
h₁ : eq a b,
h₂ : eq b c
⊢ eq a c
with_options.lean:7:36: proof state
A : Type,
a b c : A,
h₁ : eq a b,
h₂ : eq b c
⊢ eq a c
with_options.lean:7:43: proof state
A : Type,
a b c : A,
h₁ : eq a b,
h₂ : eq b c
⊢ eq a c
with_options.lean:14:16: error: invalid 'with_options' tactical, identifier (i.e., option name) expected
